docs: audit log provider (#2031)
pankod/refine: documentation/docs/core/hooks/audit-log/useLog.md

---
id: useLog
title: useLog
---

> If you need to send a create audit log **refine** provides the `useLog` hook for it, It returns the `create` or `update` method from [`auditLogProvider`](/core/providers/audit-log-provider.md#create) under the hood.

Reviewer: ```suggestion
If you need to create an audit log **refine** provides the `useLog` hook for it, It returns the `create` or `update` method from [`auditLogProvider`](/core/providers/audit-log-provider.md#create) under the hood.
```

## Usage

:::caution
This hook can only be used if `auditLogProvider`'s `create` method is provided.
:::

```tsx
import { useLog } from "@pankod/refine-core";

const { log } = useLog();

log({
    resource: "posts",
    action: "create",
    author: {
        username: "admin",
    },
    data: {
        id: 1,
        title: "New post",
    },
    meta: {
        id: 1,
    },
});
```

## API

### Properties

| Property                                                                                            | Description                                              | Type                                                              | Default |
| --------------------------------------------------------------------------------------------------- | -------------------------------------------------------- | ----------------------------------------------------------------- | ------- |

> | <div className="required-block"><div>resource</div> <div className=" required">Required</div></div> | Resource name                                            | `string`                                                          |         |

Reviewer: ```suggestion
| <div className="required-block"><div>resource</div> <div className=" required">Required</div></div> | Resource name                                            | `string`                                                          |  Resource name that it reads from URL       |
```

| author                                                                                              | Author info                                              | `Record<string, any>`                                             |         |
| meta                                                                                                | For make to unique                                       | `Record<string, any>`                                             |         |
| data                                                                                                | Metadata query for `dataProvider`                        | [`MetaDataQuery`](/core/interfaces.md#metadataquery)              | {}      |
| previousData                                                                                        | Callback to handle all related live events of this hook. | [`(event: LiveEvent) => void`](/core/interfaces.md#livemodeprops) |         |

### Return values

| Type |
| ---- |

> | any  |

Reviewer: I think, it should be update.
Additionally, we should categorize under `create` and `rename` that APIs.
